File: 94.二叉树的中序遍历.py
#
# @lc app=leetcode.cn id=94 lang=python3
#
# [94] 二叉树的中序遍历
#

# @lc code=start
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from typing import Optional
from typing import List

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

# result = []
# def mediumTraversal(root: Optional[TreeNode]):
#     if root is None :
#         return
    
#     mediumTraversal(root.left)
#     result.append(root.val)
#     mediumTraversal(root.right)

class Solution:
    def __init__(self) -> None:
        result = []

    def mediumTraversal(self, root: Optional[TreeNode]):
        if root is None :
            return
        
        self.mediumTraversal(root.left)
        self.result.append(root.val)
        self.mediumTraversal(root.right)

    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        self.mediumTraversal(root)
        return self.result
# @lc code=end

File: 94.二叉树的中序遍历.cpp
/*
 * @lc app=leetcode.cn id=94 lang=cpp
 *
 * [94] 二叉树的中序遍历
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
#include <iostream>
using namespace std;
#include <vector>

// struct TreeNode
// {
//     int val;
//     TreeNode *left;
//     TreeNode *right;
//     TreeNode() : val(0), left(nullptr), right(nullptr) {}
//     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
//     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
// };

//vector<int> result;
class Solution
{
public:
    vector<int> result;
    void mediumTraver(TreeNode *root)
    {
        if(root == NULL)
        {
            return ;
        }
        mediumTraver(root->left);
        result.push_back(root->val);
        mediumTraver(root->right);
    }

    vector<int> inorderTraversal(TreeNode *root)
    {
        mediumTraver(root);
        return result;
    }
};
// @lc code=end
